### PR TITLE
[FLINK-28222][formats] Add sql modules for csv/json formats

### PR DESCRIPTION
**Pull request by zentol**
Use the same module structure for csv/json as for other formats for consistency, and remove a shade-plugin special case.

### CODE REVIEW

**Comment by zentol** [comment]
Alternatively we could just drop the unused sql-jar profiles and not create sql-jars at all for these formats, on the assumption that they will continue to be bundled with Flink.
But I think it could be confusing for users when they look for sql-jars and don't see csv/json.

File: flink-formats/flink-sql-csv/pom.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.apache.flink</groupId>
+		<artifactId>flink-formats</artifactId>
+		<version>1.16-SNAPSHOT</version>
+	</parent>
+
+	<artifactId>flink-sql-csv</artifactId>
+	<name>Flink : Formats : SQL Avro</name>
```
Comment (zhoulii) [inline flink-formats/flink-sql-csv/pom.xml:32]: `Flink : Formats : SQL Avro` -> `Flink : Formats : SQL Csv`

**Review by zhoulii — COMMENTED** [review]
Nice work, I left a minor comment, please take a look.

File: flink-formats/flink-sql-csv/pom.xml (hunk 1 above)
Comment (zentol) [inline flink-formats/flink-sql-csv/pom.xml:32]:
```suggestion
	<name>Flink : Formats : SQL Csv</name>
```

**Review by MartijnVisser — APPROVED** [review]
LGTM

**Pull request merged**